README.md

```diff
@@ -1,126 +1,198 @@
+        └── package.yaml        # Metadata manifest containing upstream location version
 ```
 
-A regular `package.yaml` will have the following content:
+### 3. Prepare for changes
 
-```yaml
-url: https://charts.bitnami.com/bitnami/external-dns-2.20.10.tgz # url to fetch upstream chart
-packageVersion: 00 # packageVersion of modified charts, producing a $version-$packageVersion chart. For example, if istio 1.4.7 is modified with changes, rancher produces a 1.4.700 chart version that includes the modification rancher made on top of upstream charts.
-```
+Before you can prepare for changes, `packages.yaml` must be filled with the following:
+
+- `URL` - the URL that references you chart's tgz file hosted in a helm repo
+
+- `packageVersion` - The version of the package. This is appended to your chart's version in the form `{CHART_NAME}-{VERSION}{packageVersion}.tgz` after repackaging.
 
-Here is an example of upstream chart based on git repository
+#### Example `package.yaml` (Repackaged as `chart-v0.1.200.tgz`)
 
 ```yaml
-url: https://github.com/open-policy-agent/gatekeeper.git  # Url to fetch upstream chart from git
-subdirectory: chart/gatekeeper-operator # Sub directory for helm charts in git repo
-type: git # optinal, indicate that upstream chart is from git
-commit: v3.1.0-beta.8 # the revision of git repo
-packageVersion: 00 # package version
+url: https://example.com/helm-repo/chart-v0.1.2.tgz 
+packageVersion: 00 
 ``` 
 
-### Example
+Execute the following command to fetch the upstream chart (`URL` value stored in `package.yaml`), generate a `charts` 
+directory, and prepare it with the upstream chart and changes in the patch file.
 
-Todo: Add example
+```
+make prepare CHART={CHART_NAME}
+```
 
-### Workflow
+### 4. Make changes
 
-Modifying **Rancher original** charts is the same workflow as modifying helm charts. First make changes into `charts/` and commit changes. CI will automatically upload artifacts if file contents have been changed.
+Any modifications to your upstream chart like **adding the partner label** will be done in the `charts` directory.
 
-Modifying **Rancher modified** takes extra steps, as it requires modifications to be saved into patch files so that later it can retrieve the chart based on upstream chart and patch files.
+Add the partner label and required annotations in `Chart.yaml`:
 
-The step includes:
+```yaml
+annotations:
+  catalog.cattle.io/certified: partner
+  catalog.cattle.io/namespace: {CHART_NAMESPACE}       # Your chart's name is recommended
+  catalog.cattle.io/release-name: {CHART_RELEASE_NAME} # Your chart's name is recommended
+```
 
-1. Run `make CHART={CHART_NAME} prepare`
-   
-   This prepares `charts` with the current upstream chart and current patch. 
-   
-2. Change the version in `package.yaml`. If upstream chart needs to be updated, update url to point the latest chart. `packageVersion` also needs to updated.
+Save all your changes in the patch once you are done
```

Comment:
Run `make patch CHART={CHART_NAME}` to save the changes to a `{CHART_NAME}.patch` file once you are done making changes. This file will automatically be created if any changes are detected and will be used to set up the chart on a `make prepare` in a future change.